cartographer/transform/timestamped_transform.cc

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/transform/timestamped_transform.h"
+
+namespace cartographer {
+namespace transform {
+
+TimestampedTransform Interpolate(const TimestampedTransform& start,
+                                 const TimestampedTransform& end,
+                                 const common::Time& time) {
+  CHECK(start.time <= time && time <= end.time) << "Can't interpolate to time "
+                                                << time;
+
+  if (end.time == time) {
+    return end;
+  }
+  if (start.time == time) {
+    return start;
```

Comment:
Unnecessary special cases which slow down the common case?

Reply:
done